### PR TITLE
Improve names in IR

### PR DESCRIPTION
**Pull request by zachkirsch**

### CODE REVIEW

File: packages/generation/ir/ir-types/src/types.yml [hunk 1]
```diff
@@ -7,13 +7,12 @@ types:
     docs: "A type, which is a name and a shape"
     extends: commons.WithDocs
     properties:
-      name: DeclaredTypeName
+      name: TypeName
       shape: Type
-  DeclaredTypeName:
+  TypeName:
```
Comment (dsinghvi) [inline packages/generation/ir/ir-types/src/types.yml:12]: lets not do this -- it conflicts with java poet type name

Comment (zachkirsch) [reply packages/generation/ir/ir-types/src/types.yml:12]: cool, thoughts on me changing errorname and servicename to match

File: packages/generation/ir/ir-types/src/auth.yml [hunk 2]
```diff
@@ -2,9 +2,10 @@
 
 imports:
   types: types.yml
-  commons: ./commons.yml
+  commons: commons.yml
+  http: services/http.yml
 types:
-  AuthSchemeName: string
+  AuthSchemeName: commons.StringWithAllCasings
```
Comment (dsinghvi) [inline packages/generation/ir/ir-types/src/auth.yml:8]: why does the auth scheme name need to be `camel cased/ snake cased` ?

File: packages/generation/ir/ir-types/src/types.yml (hunk 1 above)
Comment (dsinghvi) [reply packages/generation/ir/ir-types/src/types.yml:12]: like `DeclaredErrorName` ?

Comment (dsinghvi) [reply packages/generation/ir/ir-types/src/types.yml:12]: and `DeclaredServiceName` ?

File: packages/generation/ir/ir-types/src/auth.yml (hunk 2 above)
Comment (zachkirsch) [reply packages/generation/ir/ir-types/src/auth.yml:8]: yeah good point, we can just use string here and enforce a good name in the linter

File: packages/generation/ir/ir-types/src/types.yml (hunk 1 above)
Comment (zachkirsch) [reply packages/generation/ir/ir-types/src/types.yml:12]: yeah

File: packages/generation/ir/ir-types/src/auth.yml (hunk 2 above)
Comment (dsinghvi) [reply packages/generation/ir/ir-types/src/auth.yml]: cool and the name there would be UpperCamelCase just like types or error names

**Pull request merged**